### PR TITLE
Replaced regex call with std::string methods and isdigit() to avoid c…

### PR DESCRIPTION
**Pull request by russell-taylor**
…rash on some compiler versions in Linux.

### CODE REVIEW

File: src/osvr/Common/PathElementTypes.cpp [hunk 1]
```diff
@@ -89,9 +88,18 @@ namespace common {
         /// port if one isn't already specified.
         static inline std::string
         attachPortToServerIfNoneSpecified(std::string const &server, int port) {
-            auto endsWithPort = std::regex{".*:[[:digit:]]+$"};
```
Comment (godbyk) [inline src/osvr/Common/PathElementTypes.cpp]: Does replacing `[[:digit:]]` with `[0-9]` work, by chance?

The underlying problem is that gcc 4.8 doesn't have full regex support.

Comment (russell-taylor) [inline src/osvr/Common/PathElementTypes.cpp]: Tried that and it didn't work.  Thanks for the idea.

**Comment by russell-taylor** [comment]
It was the only use, I thought I pulled out the #include of it...

**Comment by russell-taylor** [comment]
Must have done that in an earlier, abandoned attempt.

**Comment by russell-taylor** [comment]
Yeah, it got removed in this request as well.  Thanks.

**Pull request merged**